include/swift/SIL/SILOpenedArchetypesTracker.h

```diff
@@ -44,12 +36,6 @@ class SILOpenedArchetypesTracker : public DeleteNotificationHandler {
   SILOpenedArchetypesTracker(SILOpenedArchetypesTracker &Tracker)
       : SILOpenedArchetypesTracker(Tracker.F, Tracker) {}
```

Comment:
@hughbe Hi, I think the original code was written by me. I had a very quick look to see why we have two constructors. If I just remove the `const` constructor, I see the following:

```
swift/lib/SILOptimizer/Utils/CheckedCastBrJumpThreading.cpp:250:10: note: in instantiation of function template specialization 'llvm::Optional<swift::BasicBlockCloner>::emplace<swift::BasicBlockCloner>' requested here
  Cloner.emplace(BasicBlockCloner(CCBBlock));
         ^
swift/include/swift/SILOptimizer/Utils/Local.h:394:7: note: candidate constructor (the implicit copy constructor) not viable: expects an l-value for 1st argument
class BasicBlockCloner : public BaseThreadingCloner {
      ^
swift/include/swift/SILOptimizer/Utils/Local.h:396:5: note: candidate constructor not viable: no known conversion from 'swift::BasicBlockCloner' to 'swift::SILBasicBlock *' for 1st argument
    BasicBlockCloner(SILBasicBlock *From, SILBasicBlock *To = nullptr,
    ^
1 error generated.
```
And this seems to be the only place where the `const` copy constructor is used. It is used implicitly here, because it is probably a part of the cloner or SILBuilder.

So, my impression is that it was introduced only to handle this case. If you can re-factor the code to make it work with just one copy constructor, I think we should be fine.

Please let me review your re-factored version before merging.